core/src/main/java/org/wildfly/extras/creaper/core/online/SslOptions.java

```diff
@@ -0,0 +1,384 @@
+            }
+
+            this.trustStoreSource = Files.asByteSource(file);
+            return this;
+        }
+
+        /**
+         * Truststore with certificate for authentication, from the classpath. Optional.
+         * The resource will be loaded by the {@code resourceLoader} class ({@link Class#getResourceAsStream(String)}).
+         *
+         * @param resourceLoader class that will be used to load the truststore from classpath
+         * @param path path to the truststore on classpath (absolute or relative to {@code resourceLoader})
+         * @throws IllegalArgumentException if the {@code resourceLoader} or {@code path} is {@code null}
+         */
+        public Builder trustStore(Class resourceLoader, String path) {
+            if (resourceLoader == null) {
+                throw new IllegalArgumentException("A class for loading the truststore must be provided.");
+            }
+            if (path == null) {
+                throw new IllegalArgumentException("A path to the truststore must be provided.");
+            }
+
+            URL url = Resources.getResource(resourceLoader, path);
+            this.trustStoreSource = Resources.asByteSource(url);
+            return this;
+        }
+
+        /** Truststore password. Optional. */
+        public Builder trustStorePassword(String trustStorePassword) {
+            this.trustStorePassword = trustStorePassword;
+            return this;
+        }
+
+        /**
+         * Changes expected type of truststore. Optional.
+         * {@link KeyStoreType#DEFAULT_TYPE JDK default} is expected by default.
+         */
+        public Builder trustStoreType(KeyStoreType trustStoreType) {
+            this.trustStoreType = trustStoreType;
+            return this;
+        }
+
+        /** Builds the final {@code SslOptions}. */
+        public SslOptions build() {
+            return new SslOptions(this);
+        }
+    }
+
+
+    /**
+     * System properties like {@literal javax.net.ssl.keyStore} are not taken into account.
+     * See {@link http://docs.oracle.com/javase/6/docs/technotes/guides/security/jsse/JSSERefGuide.html#Customization}.
+     */
+    SSLContext createSslContext() {
+        final KeyManager[] keyManagers = getKeyManagers();
+        final TrustManager[] trustManagers = getTrustManagers();
+
+        try {
+            final SSLContext sslContext = SSLContext.getInstance(this.protocol.protocolName());
+            sslContext.init(keyManagers, trustManagers, null);
+
+            return sslContext;
+        } catch (GeneralSecurityException ex) {
+            throw new IllegalStateException("Failed to create SSLContext.", ex);
+        }
+    }
+
+    private KeyManager[] getKeyManagers() {
+        // pkcs11 providers are configured externally, no keystore source required here
+        if (this.keyStoreSource == null && this.keyStoreType != KeyStoreType.PKCS11) {
```

Comment:
Could you please explain this condition?

I understand that when `keyStoreSource` is `null` and `keyStoreType` is something different from `PKCS11`, then keystore wasn't configured and `return null` is the right thing to do.

But what happens when `keyStoreSource` isn't `null` and, at the same time, `keyStoreType` is `PKCS11`? Is that an invalid state?